frontend/src/components/app/ErrorScreen.js

```diff
@@ -1,25 +1,99 @@
 import counterpart from 'counterpart';
 import React, { Component } from 'react';
-
-const OFFLINE_MESSAGE_LINE1 = 'Connection lost.';
-const OFFLINE_MESSAGE_LINE2 =
-  'There are some connection issues. ' +
-  'Check connection and try to refresh the page.';
-
+import PropTypes from 'prop-types';
+import offlineMessages from '../../utils/offlineMessages';
+import { checkLoginRequest } from '../../api/login';
+import { PING_INTERVAL_BAD_GATEWAY } from '../../constants/Constants';
 class ErrorScreen extends Component {
   constructor(props) {
     super(props);
+    counterpart.registerTranslations('lang', offlineMessages);
+    this.state = { intervalId: null };
+  }
+
+  pingServer = async () => {
+    checkLoginRequest()
+      .then((response) => {
+        if (response && response.status === 200) {
+          window.location.reload();
+        }
+      })
+      .catch((e) => console.error('checkLoginRequest error:', e));
+  };
+
+  componentDidMount() {
+    const { errorType } = this.props;
+    if (errorType === 'badGateway') {
+      let intervalId = setInterval(this.pingServer, PING_INTERVAL_BAD_GATEWAY);
+      this.setState({ intervalId: intervalId });
+    }
+  }
+
+  componentWillUnmount() {
+    const { intervalId } = this.state;
+    intervalId && clearInterval(intervalId);
   }
+
+  getLineFromCounterpart = (targetLine) => {
+    const { errorType } = this.props;
+    switch (errorType) {
+      case 'badGateway':
+        return counterpart.translate(`window.error.badGateway.${targetLine}`);
+      default:
+        return counterpart.translate(`window.error.noStatus.${targetLine}`);
+    }
+  };
+
+  checkNavigatorLanguage = () => {
+    const navigatorLanguage = navigator.language;
+    if (['de', 'de-CH', 'de-AT', 'de-LU', 'de-LI'].includes(navigatorLanguage))
```

Comment:
we have an example mobile UI:
![image](https://user-images.githubusercontent.com/1244701/145053298-daeaa535-76d0-4ea0-8284-47c082a0a73e.png)

Reply:
![Screenshot 2021-12-08 at 10 13 01](https://user-images.githubusercontent.com/1708561/145172524-3ca7312e-c708-4083-9f11-55f77e2e20e6.png)

I believe we can just use what we already have instead of adding getBrowserLanguage.
![Screenshot 2021-12-08 at 10 13 01](https://user-images.githubusercontent.com/1708561/145172602-0a9f538c-f513-40b7-a1f8-cf14292518cd.png)